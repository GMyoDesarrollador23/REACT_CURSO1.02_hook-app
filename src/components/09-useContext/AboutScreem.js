import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreem = () => {
   const { user, setUser } = useContext(UserContext);

   const hanleClick = () => {
      setUser({});
   };
   return (
      <div>
         <h1>About Screen</h1>
         <hr />
         <pre>{JSON.stringify(user, null, 3)}</pre>
         <button
            onClick={hanleClick}
            className="btn btn-outline-warning"
         >
            Logout
         </button>
      </div>
   );
};
