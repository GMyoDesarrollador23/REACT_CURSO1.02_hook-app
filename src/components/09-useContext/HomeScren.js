import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScren = () => {
   const { user } = useContext(UserContext);

   return (
      <div>
         <h1>Home Screen</h1>
         <hr />
         <pre>{JSON.stringify(user, null, 3)}</pre>
      </div>
   );
};
