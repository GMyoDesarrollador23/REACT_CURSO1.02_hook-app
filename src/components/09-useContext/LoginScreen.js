import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
   const { setUser } = useContext(UserContext);

   return (
      <div>
         <h1>Login Screen</h1>
         <hr />

         <button
            className="btn btn-outline-primary"
            onClick={() =>
               setUser({
                  id: 123,
                  name: "Gerardo",
               })
            }
         >
            Login
         </button>
      </div>
   );
};
