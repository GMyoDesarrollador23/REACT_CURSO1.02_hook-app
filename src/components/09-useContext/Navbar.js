import React from "react";
import { NavLink } from "react-router-dom";

export const Navbar = () => {
   return (
      //   <nav>
      //      <ul>
      //         <li>
      //             <Link to="/">Home</Link>
      //         </li>
      //         <li>
      //             <Link to="./about">about</Link>
      //         </li>
      //         <li>
      //             <Link to="./login">Login</Link>
      //         </li>
      //      </ul>
      //   </nav>

      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
         <NavLink className="navbar-brand" to="/">
            Use Context
         </NavLink>

         <div
            className="collapse navbar-collapse"
            id="navbarNavAltMarkup"
         >
            <div className="navbar-nav">
               <NavLink
                  exact
                  activeClassName="active"
                  className="nav-item nav-link"
                  to="/"
               >
                  Home
               </NavLink>

               <NavLink
                  exact
                  activeClassName="active"
                  className="nav-item nav-link"
                  to="./about"
               >
                  About
               </NavLink>
               
               <NavLink
                  exact
                  activeClassName="active"
                  className="nav-item nav-link"
                  to="./login"
               >
                  Login
               </NavLink>
            </div>
         </div>
      </nav>
   );
};
