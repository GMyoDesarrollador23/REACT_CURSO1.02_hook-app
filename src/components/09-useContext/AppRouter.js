import React from "react";
import {
   BrowserRouter as Router,
   Switch,
   Route,
   Redirect,
} from "react-router-dom";

import { Navbar } from "./Navbar";
import { AboutScreem } from "./AboutScreem";
import { LoginScreen } from "./LoginScreen";
import { HomeScren } from "./HomeScren";

export const AppRouter = () => {
   return (
      <Router>
         <div>
            <Navbar />

            <div className="container">
               <Switch>
                  <Route
                     exact
                     path="/"
                     component={HomeScren}
                  />
                  <Route
                     exact
                     path="/about"
                     component={AboutScreem}
                  />
                  <Route
                     exact
                     path="/login"
                     component={LoginScreen}
                  />

                  <Redirect to="/" />
               </Switch>
            </div>
         </div>
      </Router>
   );
};
