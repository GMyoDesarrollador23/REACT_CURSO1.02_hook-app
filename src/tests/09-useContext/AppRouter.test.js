import React from "react";
import "@testing-library/jest-dom";
import { AppRouter } from "../../components/09-useContext/AppRouter";
import { mount } from "enzyme";
import { UserContext } from "../../components/09-useContext/UserContext";

describe("Pruebas sobre el <AppRouter/>", () => {
   const user = {
      id: 1,
      name: "Gerardo",
   };

   const wrapper = mount(
      <UserContext.Provider value={{ user }}>
         <AppRouter />
      </UserContext.Provider>
   );

   test("debe de mostrarse correctamente", () => {
      expect(wrapper).toMatchSnapshot();
   });
});
