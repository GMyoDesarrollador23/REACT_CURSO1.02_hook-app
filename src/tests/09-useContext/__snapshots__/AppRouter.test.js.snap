// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <AppRouter/> debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Use Context
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Gerardo"
}
      </pre>
    </div>
  </div>
</div>
`;
